src/order/filter/at_top_bot.lean

```diff
@@ -1168,6 +1168,42 @@ map_at_top_eq_of_gc (λb, b * k + (k - 1)) 1
     calc b = (b * k) / k : by rw [nat.mul_div_cancel b hk]
       ... ≤ (b * k + (k - 1)) / k : nat.div_le_div_right $ nat.le_add_right _ _)
 
+/-- A monotone sequence of natural numbers either tends to infinity *or* is eventually constant. -/
+lemma monotone.tendsto_at_top_at_top_or_eventually_const_nat
+  {f : ℕ → ℕ} (hf : monotone f) : tendsto f at_top at_top ∨ ∃ x, f =ᶠ[at_top] (λ _, x) :=
```

Comment:
Also, this should be `f  : ι → ℕ`, but I couldn't make `eventually_at_top` work, even with the correct typeclasses on `ι`. Not sure what's wrong.